src/main/asciidoc/_chapters/protobuf.adoc

```diff
@@ -148,3 +148,67 @@ consider extending it also in
 Going forward, we will provide a new module of common types for use
 by CPEPs that will have the same guarantees against change as does our
 public API. TODO.
+
+=== protobuf changes for hbase-3.0.0 (HBASE-23797)
+Since hadoop also shades protobuf and bumps the version to 3.x, there is no
+reason for us to stay on protobuf 2.5.0 any more.
+
+In 3.0.0, the hbase-protocol module has been purged, the CPEP implementation
+should use the protos in hbase-protocol-shaded module, and also make use of
+the shaded protobuf in hbase-thirdparty.
+
+Add this dependency to your pom:
+[source,xml]
+----
+<dependency>
+  <groupId>org.apache.hbase.thirdparty</groupId>
+  <artifactId>hbase-shaded-protobuf</artifactId>
+  <!-- use the version that your target hbase cluster uses -->
+  <version>${hbase-thirdparty.version}</version>
+  <scope>provided</scope>
+</dependency>
+----
+
+And typically you also need to add this plugin to your pom to make your
+generated protobuf code also use the shaded and relocated protobuf version
+in hbase-thirdparty.
```

Comment:
Refer users to the hbase-protoocol-shaded module? It does this? Refer them as an example to copy?

Reply:
Is there an example updated CPEP that does this? Refer to it in this doc if so?

Reply:
There is an example in hbase-examples module.

Reply:
Good. Point there? Thanks.